lib/matplotlib/pyplot.py

```diff
@@ -980,7 +980,9 @@ def savefig(*args, **kwargs):
 def figlegend(*args, **kwargs):
     return gcf().legend(*args, **kwargs)
 if Figure.legend.__doc__:
-    figlegend.__doc__ = Figure.legend.__doc__.replace("legend(", "figlegend(")
+    figlegend.__doc__ = Figure.legend.__doc__.replace(
+        "fig.legend(", "figlegend("
+    )
```

Comment:
Maybe slightly better:

```suggestion
    figlegend.__doc__ = Figure.legend.__doc__ \
        .replace("fig.legend(", "plt.figlegend(") \
        .replace("legend(", "figlegend(") \
        .replace("ax.plot(", "plt.plot(")
```

That way you can still have
-  `fig.legend(` in code examples and 
-  `legend(` for signature or mentions in code.

without any changes to the docstring.

Both of which are meaningfully translated. You also have a translation for the single-Axes examples. Only the two-Axes examples remain as a mixture of Axes and pyplot calls, but they are more difficult to handle in pyplot anyway.

Reply:
This will result in `plt.figfiglegend`
As the result of the first replacement is fed into the second replacement.

I was trying to avoid reaching for re to be able to avoid that case, but may have to.

Reply:
Actually, I think we can do `fig.legend(` -> `plt.legend(` ->`plt.figlegend(`

And it will work out, just a little confusing as the intermediate looks valid but semantically different.

Reply:
I think you can get away without awkward state and without regex. Essentially, you have two cases:

- usage in code examples -> `replace("fig.legend(", "plt.figlegend(")`
- the call signature block -> can be left as is with a bare `legend(...)` or if you want to change it, include a leading space `replace(" legend(", " plt.legend(")`

Reply:
Updated